test/echo_server/server_test.clj

```diff
@@ -0,0 +1,16 @@
+(ns echo_server.server-test
+  (:require [clojure.test :refer :all]
+            [echo_server.server :refer :all])
+  (:import (java.io InputStreamReader BufferedReader BufferedWriter OutputStreamWriter)
+           (java.net ServerSocket Socket)))
+
+(require '[clojure.core.async :as async])
+
+(deftest server-test
+  (testing "server opens a socket on localhost and listens using given port"
+    (let [port 5000]
+      (async/go (server port))
+      (async/timeout 5000)
```

Comment:
What does `async/timeout` do?

Reply:
creates a 5 second delay. this test would fail when running on travis because the client socket on the server was being created before the server socket was open and waiting. adding in a 5 second delay solved this problem.

Reply:
Ideally, the test would wait until either the server is ready or for some upper bound (in this case 5 second), whichever happens first.